### PR TITLE
Replace ssr with image card edit component

### LINKED ISSUE #331: [M3] Components - Image Card
## Feature description

Update Image Card component to match M3 style

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Material image card block is replaced with m3 specs repurposing Material Card component (https://www.figma.com/community/file/1035203688168086460)
* When the user opens Gutenberg page with m2 component it automatically migrates to M3
## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by PatelUtkarsh**
## Summary

Fixes #331 

## Checklist

- [x] My pull request is addressing an [open issue](https://github.com/material-components/material-design-for-wordpress/issues) (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/material-components/material-design-for-wordpress/blob/main/contributing/engineering.md#tests).
- [x] My code follows the [Contributing Guidelines](https://github.com/material-components/material-design-for-wordpress/blob/main/contributing.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: plugin/php/class-block-types.php [hunk 1]
```diff
@@ -245,6 +245,7 @@ public function enqueue_block_editor_assets() {
 			'postTypes'                 => $post_types,
 			'doesRequireBackCompatList' => version_compare( get_bloginfo( 'version' ), '5.8', '<' ),
 			'canUseQueryLoop'           => version_compare( '5.8', get_bloginfo( 'version' ), '<=' ),
+			'fallBackImageCard'         => $this->plugin->asset_url( 'assets/images/placeholder-image-card.png' ),
```
Comment (emeaguiar) [inline plugin/php/class-block-types.php:248]: 🥜 **Nitpick:** I'd prefer if we transform this image to either `webp` or at the very least `jpg`. I don't see any reason for this to be png and they are usually much larger that they need to be

Comment (emeaguiar) [inline plugin/php/class-block-types.php:248]: 🥜 **Nitpick:** I'd prefer if we transform this image to either `webp` or at the very least `jpg`. I don't see any reason for this to be png and they are usually much larger than they need to be

File: plugin/php/template-parts/blocks/image-card-query.php [hunk 2]
```diff
@@ -28,7 +28,8 @@
 if ( has_post_thumbnail( $post_id_image_card ) ) {
 	$thumbnail = get_the_post_thumbnail_url( $post_id_image_card );
 } else {
-	$thumbnail = get_template_directory_uri() . '/assets/images/placeholder.png';
+	global $material_design_plugin;
```
Comment (emeaguiar) [inline plugin/php/template-parts/blocks/image-card-query.php:31]: ❓ **Question**: Don't we have a `get_plugin_instance` for this? I realize it does the same but I think we don't need to explicitly call globals in code

**Review by emeaguiar — DISMISSED** [review]
Hi @PatelUtkarsh ! I left a few comments but none of them are blockers, I'll leave this as approved and you can either address them or merge if you think we can go on without them

File: plugin/php/class-block-types.php (hunk 1 above)
Comment (PatelUtkarsh) [reply plugin/php/class-block-types.php:248]: Yes generally PNG files are large, but this placeholder image is 4kb - which I've compressed down to 2kb.

File: plugin/php/template-parts/blocks/image-card-query.php (hunk 2 above)
Comment (PatelUtkarsh) [reply plugin/php/template-parts/blocks/image-card-query.php]: Good call, Updated.

File: plugin/assets/src/block-editor/blocks/card-image/edit.js [hunk 3]
```diff
@@ -20,48 +20,82 @@
...
+// eslint-disable-next-line import/no-unresolved
+import { useEntityProp, store as coreStore } from '@wordpress/core-data';
 import { useBlockProps } from '@wordpress/block-editor';
-import ServerSideRender from '@wordpress/server-side-render';
 
 /**
  * Internal dependencies
  */
-import { name } from './block.json';
+import { useSelect } from '@wordpress/data';
```
[the remark this answers is not in the document]
Comment (PatelUtkarsh) [reply plugin/assets/src/block-editor/blocks/card-image/edit.js]: Updated. :)

**Comment by PatelUtkarsh** [comment]
@emeaguiar This requires re-review, Can you check?

**Review by emeaguiar — APPROVED** [review]

**Pull request merged**